tests/components/directv/test_config_flow.py

```diff
@@ -14,219 +13,281 @@
     RESULT_TYPE_FORM,
 )
 from homeassistant.helpers.typing import HomeAssistantType
-from homeassistant.setup import async_setup_component
 
-from tests.common import MockConfigEntry
 from tests.components.directv import (
     HOST,
+    MOCK_SSDP_DISCOVERY_INFO,
+    MOCK_USER_INPUT,
     RECEIVER_ID,
-    SSDP_LOCATION,
     UPNP_SERIAL,
-    MockDirectvClass,
+    mock_connection,
+    setup_integration,
 )
+from tests.test_util.aiohttp import AiohttpClientMocker
 
 
-async def async_configure_flow(
-    hass: HomeAssistantType, flow_id: str, user_input: Optional[Dict] = None
-) -> Any:
-    """Set up mock DirecTV integration flow."""
-    with patch(
-        "homeassistant.components.directv.config_flow.DIRECTV", new=MockDirectvClass,
-    ):
-        return await hass.config_entries.flow.async_configure(
-            flow_id=flow_id, user_input=user_input
-        )
+async def test_show_user_form(hass: HomeAssistantType) -> None:
+    """Test that the user set up form is served."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={CONF_SOURCE: SOURCE_USER},
+    )
 
+    assert result["step_id"] == "user"
+    assert result["type"] == RESULT_TYPE_FORM
 
-async def async_init_flow(
-    hass: HomeAssistantType,
-    handler: str = DOMAIN,
-    context: Optional[Dict] = None,
-    data: Any = None,
-) -> Any:
-    """Set up mock DirecTV integration flow."""
-    with patch(
-        "homeassistant.components.directv.config_flow.DIRECTV", new=MockDirectvClass,
-    ):
-        return await hass.config_entries.flow.async_init(
-            handler=handler, context=context, data=data
-        )
+
+async def test_show_ssdp_confirm_form(hass: HomeAssistantType) -> None:
+    """Test that the ssdp confirmation form is served."""
+    flow = DirecTVConfigFlow()
+    flow.hass = hass
+    flow.context = {CONF_SOURCE: SOURCE_SSDP}
+    flow.discovery_info = {CONF_NAME: HOST}
+
+    result = await flow.async_step_ssdp_confirm()
```

Comment:
Please always use the core interfaces to start the tests. Use `hass.config_entries.flow.async_init`.

Reply:
should be resolved now